rtl8720dn/rpc_util.go

```diff
@@ -0,0 +1,107 @@
+package rtl8720dn
+
+import (
+	"fmt"
+	"io"
+	"time"
+)
+
+var (
+	headerBuf            [4]byte
+	readBuf              [4]byte
+	startWriteMessageBuf [1024]byte
+	payload              [1024 + 256]byte
+)
+
+const (
+	xVersion = 1
+)
+
+func startWriteMessage(msgType, service, requestNumber, sequence uint32) []byte {
+	startWriteMessageBuf[0] = byte(msgType)
+	startWriteMessageBuf[1] = byte(requestNumber)
+	startWriteMessageBuf[2] = byte(service)
+	startWriteMessageBuf[3] = byte(xVersion)
+
+	startWriteMessageBuf[4] = byte(sequence)
+	startWriteMessageBuf[5] = byte(sequence >> 8)
+	startWriteMessageBuf[6] = byte(sequence >> 16)
+	startWriteMessageBuf[7] = byte(sequence >> 24)
+
+	return startWriteMessageBuf[:8]
+}
+
+func (r *RTL8720DN) performRequest(msg []byte) error {
+	crc := computeCRC16(msg)
+	headerBuf[0] = byte(len(msg))
+	headerBuf[1] = byte(len(msg) >> 8)
+	headerBuf[2] = byte(crc)
+	headerBuf[3] = byte(crc >> 8)
+
+	if r.debug {
+		fmt.Printf("tx : %2d : ", len(headerBuf))
+		dumpHex(headerBuf[:])
+		fmt.Printf("\r\n")
+	}
+
+	r.port.Write(headerBuf[:])
+
+	if r.debug {
+		fmt.Printf("tx : %2d : ", len(msg))
+		dumpHex(msg)
+		fmt.Printf("\r\n")
+	}
+	r.port.Write(msg)
+
+	return nil
+}
+
+func dumpHex(b []byte) {
+	for i := range b {
+		if i == 0 {
+			fmt.Printf("%02X", b[i])
+		} else {
+			fmt.Printf(" %02X", b[i])
+		}
+	}
+}
+
+func (r *RTL8720DN) readThread() {
+	for {
+		n, _ := io.ReadFull(r.port, readBuf[:4])
+		if n == 0 {
+			continue
+		}
+
+		if r.debug {
+			fmt.Printf("rx : %2d : ", n)
+			dumpHex(readBuf[:n])
+			fmt.Printf("\r\n")
+		}
+
+		length := uint16(readBuf[0]) + uint16(readBuf[1])<<8
+		crc := uint16(readBuf[2]) + uint16(readBuf[3])<<8
+		//fmt.Printf("len %d (%X) crc %04X\r\n", length, length, crc)
```

Comment:
Should these commented out lines be removed or perhaps inside of `if r.debug` statements?